docs/kubernetes/getstarted.md

```diff
@@ -0,0 +1,604 @@
+# Kubernetes cluster provisioning, deploiement and examples on Threefold Grid
+
+In this guide we will walk you through the provisioning of a full-blown kubernetes cluster
+on the TF grid.
+
+We will then see how to connect to it and interact using kubectl on our local machine.
+
+Finally we will go through some examples use cases to grasp the features offered by the cluster.
+
+# TL;DR
+
+If you quickly want to setup a kubernetes cluster and start to play with it, you can launch the startup script.
+
+Prerequisite:
+
+- linux based OS
+- [kubectl](https://kubernetes.io/fr/docs/tasks/tools/install-kubectl/)
+- github account with [ssh key linked to your account](https://help.github.com/en/enterprise/2.17/user/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+
+```
+$ chmod +x startup.sh
+$ ./startup.sh zgorizzo69 "173.40" 1 2
+```
+
+This will build tfuser provision a network
+
+# Provisioning and Deploiement of the cluster
+
+Our CLI tool tfuser makes it easy to interact with the TFGrid and provision a kubernetes cluster.
+
+## Tfuser Install
+
+Tfuser is included in our zos repository. Let's clone the repository and build the binary
+
+```
+$ git clone https://github.com/threefoldtech/zos
+$ cd tools
+$ make tfuser
+cd tfuser && go build -ldflags '-w -s -X github.com/threefoldtech/zos/pkg/version.Branch=fcvm -X github.com/threefoldtech/zos/pkg/version.Revision=c00740b8fbb75746cd02fb5614ccf57342b7a9f4 -X github.com/threefoldtech/zos/pkg/version.Dirty='
+$ cd tfuser
+$ ./tfuser --help
+```
+
+## Identity on the TF Grid
+
+To interact with the grid we need an indentity which is simply some cryptographic keys that we will use to sign transaction on the network.
+Let's generate them with tfuser
+
+```
+$ tfuser id
+new identity generated: 7HZd49ugHzJxQUnHKyQRbmcwWWvKKtoJUndfiiXGTkow
+seed saved at user.seed
+```
+
+NB: a file named user.seed has been created. This file is sensitive as it stores your private key. Make sure not to loose that file.
+
+## Provisioning
+
+To provision our VMs we first need to setup a network between different Nodes on the TFGrid
+We will use devnet Nodes to select some nodes go to :[devnet Cockpit](https://cockpit.devnet.grid.tf/)
```

Comment:
If they want an API to explore the nodes on devnet, they can use https://explorer.devnet.grid.tf/nodes

Reply:
comment added